transaction-status/src/extract_memos.rs

```diff
@@ -0,0 +1,40 @@
+use {
+    crate::parse_instruction::parse_memo_data,
+    solana_sdk::{message::Message, pubkey::Pubkey},
+};
+
+// A helper function to convert spl_memo::v1::id() as spl_sdk::pubkey::Pubkey to
+// solana_sdk::pubkey::Pubkey
+pub fn spl_memo_id_v1() -> Pubkey {
+    Pubkey::new_from_array(spl_memo::v1::id().to_bytes())
+}
+
+// A helper function to convert spl_memo::id() as spl_sdk::pubkey::Pubkey to
+// solana_sdk::pubkey::Pubkey
+pub fn spl_memo_id_v3() -> Pubkey {
+    Pubkey::new_from_array(spl_memo::id().to_bytes())
+}
+
+pub fn extract_and_fmt_memos(message: &Message) -> Option<String> {
+    let memos = extract_memos(message);
+    if memos.is_empty() {
+        None
+    } else {
+        Some(memos.join("; "))
```

Comment:
I'm a little nervous about somebody wanting to use `;` in a memo and then running into parsing issues, especially since `;` is a URL field delimiter.

What we if additionally prefixed each memo by the UTF8 string length:
* `[12] Single memo`
* `[4] Test; [26] This is a test second memo`
The readability is mostly maintained while making the string much easier to correctly parse programatically.

Reply:
Nice idea: https://github.com/solana-labs/solana/pull/19512/commits/519a002bb305e06c9594e0861e2e551e056ed465

```
$ solana transaction-history F8vKjn13shKnef6pGfoHuWfpWP6DqmWT3xZK7oahnGuw -v
RPC URL: http://localhost:8899
Default Signer Path: id.json
Commitment: confirmed
2kViiexw14Qi3NwpskeqHewJJoYGjoqynj195Y9Uc9MBN1iARJjn2Rt1Yuuean4Lxp1S8v6xADgGSgPFWTDdFiwN [slot=132 timestamp=2021-08-31T01:15:33Z status=Finalized] [4] 🦖; [26] This is a test second memo
```